firmware/library/L1_Drivers/timer.hpp

```diff
@@ -0,0 +1,157 @@
+//                  MAT 0     , Mat 1     , Mat 2     , Mat 3
+
+  explicit constexpr Timer(TimerPort mode)
+                      : channel_(static_cast<uint8_t>(mode)),
+                        mat0_(&quads[channel_][0]),
+                        mat1_(&quads[channel_][1]),
+                        mat2_(&quads[channel_][2]),
+                        mat3_(&quads[channel_][3])
+  {
+  }
+
+  void Initialize(TimerPort timer) override
+  {
+    PowerUpPeripheral(Lpc40xxSystemController::Peripherals::kTimer0);
+    PowerUpPeripheral(Lpc40xxSystemController::Peripherals::kTimer1);
+    PowerUpPeripheral(Lpc40xxSystemController::Peripherals::kTimer2);
+    PowerUpPeripheral(Lpc40xxSystemController::Peripherals::kTimer3);
+    mat0_->SetPinFunction(kTime0Mat0);
+    mat0_->SetAsAnalogMode(true);
+    mat0_->SetMode(PinInterface::Mode::kInactive);
+    mat1_->SetPinFunction(kTime0Mat0);
+    mat1_->SetAsAnalogMode(true);
+    mat1_->SetMode(PinInterface::Mode::kInactive);
+    mat2_->SetPinFunction(kTime0Mat0);
+    mat2_->SetAsAnalogMode(true);
+    mat2_->SetMode(PinInterface::Mode::kInactive);
+    mat3_->SetPinFunction(kTime0Mat0);
+    mat3_->SetAsAnalogMode(true);
+    mat3_->SetMode(PinInterface::Mode::kInactive);
+    // set Prescale register for Prescale Counter to milliseconds
+    tim_register[static_cast<uint8_t>(timer)]->PR &= ~(kClear << 1);
+    tim_register[static_cast<uint8_t>(timer)]->PR |= (kPrescaleMicro << 1);
+    tim_register[static_cast<uint8_t>(timer)]->TCR |= (1 << 0);
+  }
+
+  // Function that sets a timer of your choice (0-3) to a specific
+  // time in milliseconds,and designates the mode of
+  // functionality you want from it, interrupt, stop, or reset.
+  // --NOTICE-- Currently Each Timer is directly hooked to match
+  void SetTimer(uint32_t time, RegisterChoice reg, TimerSetting mode) override
+  {
+        switch (reg)
+    {
+        case kRegMR0:
+            tim_register[channel_]->MCR &= ~(kClearMode << kMat0);
+            tim_register[channel_]->MCR |= (mode << kMat0);
+            tim_register[channel_]->MR0 |= (((time/2)*1000) << 0);
+            break;
+        case kRegMR1:
+            tim_register[channel_]->MCR &= ~(kClearMode << kMat1);
+            tim_register[channel_]->MCR |= (mode << kMat1);
+            tim_register[channel_]->MR1 |= (((time/2)*1000) << 0);
+            break;
+        case kRegMR2:
+            tim_register[channel_]->MCR &= ~(kClearMode << kMat2);
+            tim_register[channel_]->MCR |= (mode << kMat2);
+            tim_register[channel_]->MR2 |= (((time/2)*1000) << 0);
+            break;
+        case kRegMR3:
+            tim_register[channel_]->MCR &= ~(kClearMode << kMat3);
+            tim_register[channel_]->MCR |= (mode << kMat3);
+            tim_register[channel_]->MR3 |= (((time/2)*1000) << 0);
+            break;
+        default:
+            break;
+    }
+  }
+
+  uint32_t GetTimer() override
```

Comment:
```suggestion
  [[gnu::always_inline]] uint32_t GetTimer() override
```
This will optimize this function to force it to become unlined at the call site, rather than doing an actual function call.